Clarify NOTE_STATE_CHANGE_EVENT documentation with HTML links and parameter references (#722)
canvas-medical/documentation: collections/_sdk/events.md

#### Notes

> 

Reviewer: ```suggestion
```

<table>
  <thead>
    <tr><th colspan="2">NOTE_STATE_CHANGE_EVENT_CREATED</th></tr>

>     <tr><td colspan="2">Occurs as a note traverses through its state machine. This event can be used when looking at any changes to the [note state](/sdk/data-note/#notestates), including locking and unlocking.</td></tr>

Reviewer: @copilot please change this to be a html link and not markup (I assume that's why it's not working as expected)

"type": NoteStateChangeEvent</pre></td>
      <td><pre>"note_id": note_id,
"patient_id": pt_id,

> "state": <a href="https://docs.canvasmedical.com/sdk/data-note/#notestates">str</a></pre></td>

Reviewer: Add this same link to the "str" on line 1360

Reviewer: @copilot please take this action

> "state": <a href="https://docs.canvasmedical.com/sdk/data-note/#notestates">str</a></pre></td>

Reviewer: ```suggestion
"state": <a href="/sdk/data-note/#notestates">str</a></pre></td>
```

<table>
  <thead>
    <tr><th colspan="2">NOTE_STATE_CHANGE_EVENT_UPDATED</th></tr>

>     <tr><td colspan="2">Occurs if a note state change event is updated. This event should be used when the note is locked AND when user needs the PDF. Note state change events are updated when a note is locked and we generate an archived PDF copy of the note; this is done asynchronously.</td></tr>

Reviewer: ```suggestion
    <tr><td colspan="2">Occurs if a note state change event is updated. Locking and unlocking both trigger an update event, and there is an *additional* update event when an archived PDF copy of the note finishes generating; this is done asynchronously.</td></tr>
```

"type": NoteStateChangeEvent</pre></td>
      <td><pre>"note_id": note_id,
"patient_id": pt_id,

> "state": <a href="https://docs.canvasmedical.com/sdk/data-note/#notestates">str</a></pre></td>

Reviewer: ```suggestion
"state": <a href="/sdk/data-note/#notestates">str</a></pre></td>
```
